src/music/MusicManager.ts

```diff
@@ -0,0 +1,185 @@
+import { Note } from "./Notes.js";
+import Instrument from "./Instrument.js";
+import { AdsrOscillatorInstrument } from "./Instruments.js";
+import { Random } from "../util/Random.js";
+import { Scales, Scale, ScaleQuery } from "./Scales.js";
+import Rhythm from "./Rhythm.js";
+import { Drumkit, Drums } from "./Drums.js";
+import { Arrays, NonEmptyArray } from "../util/Arrays.js";
+
+export namespace MusicManager {
+
+    export const context: AudioContext = new AudioContext();
+
+    export const instruments = {
+        arp: new AdsrOscillatorInstrument(
+            { type: "sawtooth", detune: 3 },
+            { attack: 0.01, decay: 0.05, sustain: 0.0, release: 0.1 }
+        ),
+        bass: new AdsrOscillatorInstrument(
+            { type: "triangle", detune: 2 },
+            { attack: 0.01, decay: 0.01, sustain: 1.0, release: 0.1 }
+        ),
+        pad: new AdsrOscillatorInstrument(
+            { type: "triangle", detune: 4 },
+            { attack: 1.0, decay: 0.1, sustain: 0.5, release: 0.5 }
+        )
+    };
+
+    export const masterGain: GainNode = context.createGain();
+    export const drumkit: Drumkit = new Drumkit();
+
+    interface MusicState {
+        beatsPerMinute: number;
+        rhythm: Rhythm;
+        root: number;
+        scale: Scale;
+    }
+
+    const state: MusicState = {
+        beatsPerMinute: 220,
+        rhythm: new Rhythm(),
+        root: 60, // middle C
+        scale: 2741 // major
+    };
+
+    enum ChordFunction {
+        TONIC = "tonic",
+        SUBDOMINANT = "subdominant",
+        DOMINANT = "dominant",
+        AMBIGUOUS = "ambiguous"
+    }
```

Comment:
Why is this nested in a namespace and not top-level?

Reply:
Chord functions don't seem useful outside of `MusicManager`, but I can be convinced otherwise.